dartagnan/src/main/java/com/dat3m/dartagnan/wmm/definition/AllocPtr.java

```diff
@@ -0,0 +1,18 @@
+package com.dat3m.dartagnan.wmm.definition;
+
+import com.dat3m.dartagnan.wmm.Definition;
+import com.dat3m.dartagnan.wmm.Relation;
+
+import static com.dat3m.dartagnan.wmm.RelationNameRepository.ALLOCPTR;
+
+public class AllocPtr extends Definition {
+
+    public AllocPtr(Relation r) {
+        super(r, ALLOCPTR);
+    }
+
+    @Override
+    public <T> T accept(Visitor<? extends T> v) {
+        return v.visitAllocPtr(this);
+    }
+}

```

Comment:
In your current iteration, `Alloc` and `Free` have `loc`-edges (which I think they should), so `let allocPtr = [ALLOC];loc;[FREE]` holds and you can save yourself this whole relation.